src/main/java/logic/App.java

```diff
@@ -130,28 +132,24 @@ private void addReference(String i) {
         inputFields(reference);
         addRefToList(reference, io, references);
     }
-    
+
     private void inputFields(Reference reference) {
         for (String inputField : reference.getRequiredFields()) {
             String inputLine;
 
             io.print(inputField + ": ");
             inputLine = io.readLine();
 
-            if (scandeja(inputLine)
-                    || (numerics.contains(inputField)
-                    && !inputLine.matches("[0-9]+"))
-                    || inputLine.isEmpty()) {
+            if (!reference.setField(inputField, inputLine, validator)
```

Comment:
Tätä jäin miettimään sillon päivällä kun puhuttiin sprinttidemon jälkeen. Oisko tän voinu tehdä yksinkertasemmin niin, että validator-luokalla on omat tarkastusmetodit (edit: tai siis geneerisyyden säilyttämiseks ehkä parempi alla olevalla tavalla - yks metodi) jokaiselle eri kentälle. Kutsuminen menis tän rivin koodin sijaan suunnilleen seuraavasti:

`if (!reference.setField(inputField, inputLine) || (inputLine.length() == 0) || !validator.checkCorrectness(inputField, inputLine) ({`
'     
Sillon ei tarvis välttämättä interfacea ja pikkuluokkia tehdä? (en ehtinyt tsekkaamaan, että tuoko ne muuta funktionaalisuutta, mutta lisää kompleksisuutta). Ja sit tuolla validator-luokassa olis noi tarkistukset vaikka just switch case-tyylisesti inputFieldin mukaan.